lib/srv/discovery/discovery_test.go

```diff
@@ -422,35 +413,20 @@ func TestDiscoveryServer(t *testing.T) {
 				require.NoError(t, r.Close())
 				require.NoError(t, w.Close())
 			})
-			if tc.logHandler != nil {
-				logger.SetOutput(w)
-				logger.SetLevel(logrus.DebugLevel)
-			}
 
 			go server.Start()
 			t.Cleanup(server.Stop)
 
-			if tc.logHandler != nil {
-				done := make(chan struct{})
-				go tc.logHandler(t, r, done)
-				timeoutCtx, cancelfn := context.WithTimeout(ctx, time.Second*5)
-				defer cancelfn()
-				select {
-				case <-timeoutCtx.Done():
-					t.Fatal("Timeout waiting for log entries")
-					return
-				case <-done:
-				}
-			}
-
-			if tc.wantEvents > 0 {
+			if len(tc.wantInstalledInstances) > 0 {
 				require.Eventually(t, func() bool {
-					return reporter.EventsCount() == tc.wantEvents
-				}, 100*time.Millisecond, 10*time.Millisecond)
+					require.ElementsMatch(t, tc.wantInstalledInstances, installer.GetInstalledInstances())
```

Comment:
We shouldn't use require inside an Eventually, as the predicate runs in a separate goroutine and require must only be called on the test's main goroutine